src/GraphQL/Types/Scalars/StringGraphType.cs

```diff
@@ -9,15 +9,25 @@ namespace GraphQL.Types
     public class StringGraphType : ScalarGraphType
     {
         /// <inheritdoc/>
-        public override object ParseLiteral(IValue value) => (value as StringValue)?.Value;
+        public override object ParseLiteral(IValue value) => value switch
+        {
+            StringValue s => s.Value,
+            NullValue _ => null,
+            _ => ThrowLiteralConversionError(value)
+        };
 
         /// <inheritdoc/>
-        public override object ParseValue(object value) => value?.ToString();
+        public override bool CanParseLiteral(IValue value) => value is StringValue || value is NullValue;
 
         /// <inheritdoc/>
-        public override bool CanParseLiteral(IValue value) => value is StringValue;
+        public override object ParseValue(object value) => value switch
+        {
+            string _ => value,
+            null => null,
+            _ => ThrowValueConversionError(value)
+        };
```

Comment:
Note that Serialize calls ParseValue.  So strings returned from resolvers must now be strings for StringGraphType

Reply:
OK. It should be clearly stated in the documentation that in version 4 scalars became stricter on the supported types. They practically stopped doing any value conversions.